### DISCUSSION TITLE
Problems installing USBIPD

### DISCUSSION THREAD

**Discussion by Roadracerboy** [dorssel/usbipd-win#826] [Q&A] [answered]
My first question here so Im sorry if I make the question in a wrong format :-)

This is not working for me, it is something I maybe doing wrong or dont understand from the instructions in: https://github.com/dorssel/usbipd-win
I run the installer (.msi) from the [latest release](https://github.com/dorssel/usbipd-win/releases/latest) on the Windows machine where your USB device is connected.

Going in to WSL 
($ uname -r 5.15.133.1-microsoft-standard-WSL2, Linux 5.15.133.1-microsoft-standard-WSL2 #1 SMP Thu Oct 5 21:02:42 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

Trying to install, "Unable to locate package usbipd" but it rejects me and want other install first that the system cannot find. 

Stucked here, what are the steps I need to do now?

My computer:

Intel(R) Core(TM) i5-6300U CPU @ 2.40GHz   2.50 GHz
RAM-minne     8,00 GB (7,86 GB går att använda)
Systemtyp	64-bitars operativsystem, x64-baserad processor

Windows subsystem for Linux and Virtual machine platform is both activated and system rebooted

```
$ sudo apt install usbipd
Reading package lists... Done
Building dependency tree
Reading state information... Done
E: Unable to locate package usbipd
```

```
$ usbip
WARNING: usbip not found for kernel 5.15.133.1-microsoft

  You may need to install the following packages for this specific kernel:
    linux-tools-5.15.133.1-microsoft-standard-WSL2
    linux-cloud-tools-5.15.133.1-microsoft-standard-WSL2

  You may also want to install one of the following packages to keep up to date:
    linux-tools-standard-WSL2
    linux-cloud-tools-standard-WSL2
```

```
$ sudo apt install linux-tools-5.15.133.1-microsoft-standar
d-WSL2
Reading package lists... Done
Building dependency tree
Reading state information... Done
E: Unable to locate package linux-tools-5.15.133.1-microsoft-standard-WSL2
E: Couldn't find any package by glob 'linux-tools-5.15.133.1-microsoft-standard-WSL2'
```

**Comment by Roadracerboy** [comment]
I pull back my question, found out that usbipd is NOT possible with my x64-computer. Usbipd require x86.

> Prerequisites
> 

> Running Windows 11 (Build 22000 or later). (Windows 10 support is possible, see note below).
> A machine with an x64/x86 processor is required. (Arm64 is currently not supported with usbipd-win).

https://learn.microsoft.com/en-us/windows/wsl/connect-usb#prerequisites

**Reply by dorssel** [reply, marked as answer]
Wrong. x64 is correct.

Your problem is that you try to run `usbipd` in WSL. Instead, you should run it in Windows (e.g., in PowerShell).
And the documentation on Microsoft is outdated. A Pull Request to correct it is already pending here: https://github.com/MicrosoftDocs/WSL/pull/1870.

Just follow the instructions on https://github.com/dorssel/usbipd-win/wiki/WSL-support

**Reply by Roadracerboy** [reply]
Ok, thanks Dorssel! Then I go back to make this work :-)  

Tried the instructions a couple of times, learned and understand now this with the cmd-prompt when it takes admin and when not.

So, now when I binded and attached, going in Ubuntu and lsusb with good result:

```
$ lsusb (in Ubuntu)
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 001 Device 002: ID 0bda:8812 Realtek Semiconductor Corp. RTL8812AU 802.11a/b/g/n/ac 2T2R DB WLAN Adapter
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
```

(In Ubuntu) Installed Aircracks driver and the program. But the wlan0 that is supposed to show up with $ nmcli d
is not showed up. (All this works for med in VMware + Kali, but I want to set it up in Ubuntu and wsl vithout virtual machine)

```
$ nmcli d
DEVICE  TYPE      STATE      CONNECTION
eth0    ethernet  unmanaged  --
lo      loopback  unmanaged  --
```

And with airmon my device doesnt show up.
```
$ sudo airmon-ng
PHY     Interface       Driver          Chipset
```

So I hope you or somebody else can push me in the right direction to make the device show up as wlan0, was hoping usbipd was the final solution but apparently there is some more work to do :-)

**Reply by Roadracerboy** [reply]
Im sorry if my question leeds to topics ouside your expertis :-)

**Reply by Roadracerboy** [reply]
Also one question about the errormessage in CMD-prompt. 
`usbipd: warning: USB filter 'USBPcap' is known to be incompatible with this software; 'bind --force' will be required.`

Should it just be ignored? Searched and found some issues-articles here in github but it seems there is nothing to do about it.

The connection in to Ubuntu is working anyway I suppose.. rtl8812au shows up
```
$ lsusb
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 001 Device 002: ID 0bda:8812 Realtek Semiconductor Corp. RTL8812AU 802.11a/b/g/n/ac 2T2R DB WLAN Adapter
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
```

**Reply by dorssel** [reply]
If you can attach, you can ignore it. If you cannot attach, you either need to uninstall that filter (or the software it comes with), or use `bind --force` instead.

Note that the default WSL kernel does not have USB network drivers. You will need to build your own kernel as per the Wiki. Currently, `lsusb` will list the device, but the kernel doesn't have an actual driver for it, so it does not show up as a network device.

**Reply by Roadracerboy** [reply]
Ok then I ignore because it works anyway as described. Driver is installed, the same as I use in VMware/Kali, or is it still problems with the kernel? Can you link me to some reading-material and sources to start with? Or does it require a lot of programming experience?

Good at welding and constructions, but not yet programming Lol :-)

**Reply by Roadracerboy** [reply]
I see now you have a lot of information here about Building your own WSL 2 kernel with additional drivers.
https://github.com/dorssel/usbipd-win/wiki/WSL-support
